From 8ea94d26ece3115a7536f350809942938308a78b Mon Sep 17 00:00:00 2001
From: jiangcuo <jiangcuo@bingsin.com>
Date: Thu, 5 Dec 2024 12:22:17 +0800
Subject: [PATCH 4/6] bump version to 5.2.2+port1

---
 debian/changelog | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/debian/changelog b/debian/changelog
index 03b61e9..8c6b995 100644
--- a/debian/changelog
+++ b/debian/changelog
@@ -66,6 +66,14 @@ pve-container (5.2.3) bookworm; urgency=medium
 
  -- Proxmox Support Team <support@proxmox.com>  Thu, 19 Dec 2024 15:16:59 +0100
 
+pve-container (5.2.2+port1) bookworm; urgency=medium
+
+  * Add Anolis support
+
+  * Set all unknown ostype to unmanaged
+
+ -- Jiangcuo <jiangcuo@bingsin.com>  Thu, 05 Dec 2024 12:18:26 +0800
+
 pve-container (5.2.2) bookworm; urgency=medium
 
   * network config: avoid committing invalid config on hotplug failure, rather
-- 
2.39.5

