From bba618e88b694a38f63de8b4bfaa403b7e3cad2f Mon Sep 17 00:00:00 2001
From: jiangcuo <jiangcuo@bingsin.com>
Date: Thu, 5 Dec 2024 12:21:23 +0800
Subject: [PATCH] set unknown ostype to unmanaged

---
 src/PVE/LXC/Setup.pm | 2 --
 1 file changed, 2 deletions(-)

diff --git a/src/PVE/LXC/Setup.pm b/src/PVE/LXC/Setup.pm
index 98398f8..a3a4588 100644
--- a/src/PVE/LXC/Setup.pm
+++ b/src/PVE/LXC/Setup.pm
@@ -87,8 +87,6 @@ my $autodetect_type = sub {
 	return "nixos";
     } elsif (-f "$rootdir/etc/openEuler-release") {
 	return "openeuler";
-    } elsif (-f "$rootdir/etc/os-release") {
-	die "unable to detect OS distribution\n";
     } else {
 	warn "/etc/os-release file not found and autodetection failed, falling back to 'unmanaged'\n";
 	return "unmanaged";
