diff --git a/src/Utils.js b/src/Utils.js
index c873c85..8ee007f 100644
--- a/src/Utils.js
+++ b/src/Utils.js
@@ -559,24 +559,7 @@ utilities: {
 		    Ext.Msg.alert(gettext('Error'), response.htmlStatus);
 		},
 		success: function(response, opts) {
-		    let res = response.result;
-		    if (res === null || res === undefined || !res || res
-			.data.status.toLowerCase() !== 'active') {
-			Ext.Msg.show({
-			    title: gettext('No valid subscription'),
-			    icon: Ext.Msg.WARNING,
-			    message: Proxmox.Utils.getNoSubKeyHtml(res.data.url),
-			    buttons: Ext.Msg.OK,
-			    callback: function(btn) {
-				if (btn !== 'ok') {
-				    return;
-				}
-				orig_cmd();
-			    },
-			});
-		    } else {
-			orig_cmd();
-		    }
+		    orig_cmd();
 		},
 	    },
 	);
