diff --git a/debian/changelog b/debian/changelog
index 15bb2ed45..a2f207bac 100644
--- a/debian/changelog
+++ b/debian/changelog
@@ -1,3 +1,9 @@
+openssh (1:9.9p1-3+port) port; urgency=medium
+
+  *  Enable dsa for proxmox lxc
+
+ -- Jiangcuo <jiangcuo@lierfang.com>  Wed, 25 Dec 2024 10:49:24 +0800
+
 openssh (1:9.9p1-3) unstable; urgency=medium
 
   * Fix mlkem768x25519-sha256 key exchange algorithm on big-endian
diff --git a/debian/rules b/debian/rules
index d0a45d335..a5c7df9f2 100755
--- a/debian/rules
+++ b/debian/rules
@@ -79,6 +79,7 @@ confflags_udeb := $(confflags)
 
 # Options specific to the deb build.
 confflags += --with-tcp-wrappers
+confflags += --enable-dsa-keys
 confflags += --with-pam
 confflags += --with-libedit
 confflags += --with-kerberos5=/usr
