From aef90f1371569e58995a9bf2cde4d83a0dbc4d84 Mon Sep 17 00:00:00 2001
From: jiangcuo <jiangcuo@bingsin.com>
Date: Mon, 3 Feb 2025 13:53:52 +0800
Subject: [PATCH 3/9] add arch when qm list

---
 PVE/CLI/qm.pm | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/PVE/CLI/qm.pm b/PVE/CLI/qm.pm
index 4214a7ca..18ab3326 100755
--- a/PVE/CLI/qm.pm
+++ b/PVE/CLI/qm.pm
@@ -1124,11 +1124,11 @@ our $cmddef = {
 	my $vmlist = shift;
 	exit 0 if (!scalar(@$vmlist));
 
-	printf "%10s %-20s %-10s %-10s %12s %-10s\n",
-	qw(VMID NAME STATUS MEM(MB) BOOTDISK(GB) PID);
+	printf "%10s %-20s %-12s %-10s %-10s %12s %-10s\n",
+	qw(VMID NAME ARCH STATUS MEM(MB) BOOTDISK(GB) PID);
 
 	foreach my $rec (sort { $a->{vmid} <=> $b->{vmid} } @$vmlist) {
-	    printf "%10s %-20s %-10s %-10s %12.2f %-10s\n", $rec->{vmid}, $rec->{name},
+	    printf "%10s %-20s %-12s %-10s %-10s %12.2f %-10s\n", $rec->{vmid}, $rec->{name}, $rec->{arch},
 	        $rec->{qmpstatus} || $rec->{status},
 	        ($rec->{maxmem} || 0)/(1024*1024),
 	        ($rec->{maxdisk} || 0)/(1024*1024*1024),
-- 
2.39.5

