From 94c99b8c369887856767383105fe5f2a94e9a555 Mon Sep 17 00:00:00 2001
From: jiangcuo <jiangcuo@bingsin.com>
Date: Mon, 3 Feb 2025 16:20:25 +0800
Subject: [PATCH 9/9] bump version to 8.3.6+port

---
 debian/changelog | 88 ++++++++++++++++++++++++++++++++++++++++++++++++
 1 file changed, 88 insertions(+)

diff --git a/debian/changelog b/debian/changelog
index 90f24779..45c55312 100644
--- a/debian/changelog
+++ b/debian/changelog
@@ -7,6 +7,14 @@ qemu-server (8.3.7) bookworm; urgency=medium
 
  -- Proxmox Support Team <support@proxmox.com>  Tue, 04 Feb 2025 17:12:25 +0100
 
+qemu-server (8.3.6+port) bookworm; urgency=medium
+
+  * Update to Port version
+   
+  * Make the x86_64 code more compliant with the upstream version
+
+ -- Jiangcuo <Jiangcuo@lierfang.com>  Mon, 03 Feb 2025 13:56:49 +0800
+
 qemu-server (8.3.6) bookworm; urgency=medium
 
   * api: do not set an empty machine option when pinning
@@ -43,6 +51,24 @@ qemu-server (8.3.4) bookworm; urgency=medium
 
  -- Proxmox Support Team <support@proxmox.com>  Fri, 17 Jan 2025 19:32:17 +0100
 
+qemu-server (8.3.3+port3) bookworm; urgency=medium
+
+  * Add x86_64 cpu for compatible with x86 host
+
+ -- Jiangcuo <Jiangcuo@lierfang.com>  Fri, 17 Jan 2025 11:57:51 +0800
+
+qemu-server (8.3.3+port2) bookworm; urgency=medium
+
+  * Add 'gicversion' param.
+
+ -- Jiangcuo <Jiangcuo@lierfang.com>  Thu, 26 Dec 2024 12:14:33 +0800
+
+qemu-server (8.3.3+port) bookworm; urgency=medium
+
+  * Add port version
+
+ -- Jiangcuo <Jiangcuo@lierfang.com>  Thu, 26 Dec 2024 12:13:33 +0800
+
 qemu-server (8.3.3) bookworm; urgency=medium
 
   * fix #5980: import disk: fix spurious warning if no target disk is defined
@@ -57,12 +83,26 @@ qemu-server (8.3.3) bookworm; urgency=medium
 
  -- Proxmox Support Team <support@proxmox.com>  Sun, 15 Dec 2024 14:26:32 +0100
 
+qemu-server (8.3.2+port) bookworm; urgency=medium
+
+  * Add port version
+
+ -- Jiangcuo <Jiangcuo@lierfang.com>  Thu, 26 Dec 2024 11:28:33 +0800
+
 qemu-server (8.3.2) bookworm; urgency=medium
 
   * use image format from storage layer for PVE-managed volumes
 
  -- Proxmox Support Team <support@proxmox.com>  Mon, 09 Dec 2024 10:08:14 +0100
 
+qemu-server (8.3.1+port) bookworm; urgency=medium
+
+  * Set default bios type to ovmf when vm create
+   
+  * Add Kunpeng920 cpu model.
+
+ -- Jiangcuo <Jiangcuo@lierfang.com>  Thu, 26 Dec 2024 11:22:33 +0800
+
 qemu-server (8.3.1) bookworm; urgency=medium
 
   * api: clone vm: make error for unsupported volumes appear in a
@@ -72,6 +112,24 @@ qemu-server (8.3.1) bookworm; urgency=medium
 
  -- Proxmox Support Team <support@proxmox.com>  Thu, 05 Dec 2024 12:38:12 +0100
 
+qemu-server (8.3.0+port3) bookworm; urgency=medium
+
+  * Add loongarch efi_vars support
+
+ -- Jiangcuo <jiangcuo@lierfang.com>  Wed, 11 Dec 2024 13:44:37 +0800
+
+qemu-server (8.3.0+port2) bookworm; urgency=medium
+
+  * Fix usb hotplug error
+
+ -- Jiangcuo <jiangcuo@bingsin.com>  Mon, 09 Dec 2024 16:30:48 +0800
+
+qemu-server (8.3.0+port1) bookworm; urgency=medium
+
+  * Fix cpu type on loongarch64
+
+ -- Jiangcuo <jiangcuo@bingsin.com>  Thu, 05 Dec 2024 14:16:44 +0800
+
 qemu-server (8.3.0) bookworm; urgency=medium
 
   * api: import working storage: improve error message
@@ -269,6 +327,20 @@ qemu-server (8.1.2) bookworm; urgency=medium
 
  -- Proxmox Support Team <support@proxmox.com>  Fri, 19 Apr 2024 16:09:18 +0200
 
+qemu-server (8.1.1+port2) bookworm; urgency=medium
+
+  * fix query_supported_cpu_flags on aarch64
+
+ -- jiangcuo <jiangcuo@bingsin.com>  Wed, 27 Mar 2024 09:26:45 +0800
+
+qemu-server (8.1.1+port1) bookworm; urgency=medium
+
+  * update to 8.1.1
+
+  * remove qemu cpu flags check
+
+ -- jiangcuo <jiangcuo@bingsin.com>  Fri, 22 Mar 2024 21:11:29 +0800
+
 qemu-server (8.1.1) bookworm; urgency=medium
 
   * config: pending network: avoid undef-warning on old/new comparison
@@ -321,6 +393,12 @@ qemu-server (8.1.0) bookworm; urgency=medium
 
  -- Proxmox Support Team <support@proxmox.com>  Fri, 08 Mar 2024 15:00:25 +0100
 
+qemu-server (8.0.10+pve1) bookworm; urgency=medium
+
+  * add pve-edk2-firmware-aarch64 depends
+
+ -- jiangcuo <jiangcuo@bingsin.com>  Sun, 03 Dec 2023 01:46:23 -0500
+
 qemu-server (8.0.10) bookworm; urgency=medium
 
   * sdn: pass vmid and hostname to allow requesting a new mapping
@@ -339,6 +417,16 @@ qemu-server (8.0.9) bookworm; urgency=medium
 
  -- Proxmox Support Team <support@proxmox.com>  Tue, 21 Nov 2023 15:40:27 +0100
 
+qemu-server (8.0.8-1) bookworm; urgency=medium
+
+  * remove some scshw
+
+  * add new cputype
+
+  * add virt machine
+
+ -- jiangcuo <jiangcuo@bingsin.com>  Sun, 19 Nov 2023 14:29:21 +0800
+
 qemu-server (8.0.8) bookworm; urgency=medium
 
   * fix #2816: restore: remove timeout when allocating disks
-- 
2.39.5

