diff --git a/debian/control b/debian/control
index 89a11181..d989cb95 100644
--- a/debian/control
+++ b/debian/control
@@ -21,6 +21,7 @@ Build-Depends: debhelper-compat (= 13),
                pve-cluster,
                pve-doc-generator (>= 6.2-5),
                pve-edk2-firmware,
+               pve-edk2-firmware-aarch64,
                pve-firewall,
                pve-ha-manager <!nocheck>,
                pve-qemu-kvm (>= 7.1~),
@@ -51,6 +52,7 @@ Depends: dbus,
 # TODO: make legacy edk2 optional (suggests) for PVE 9 and warn explicitly about it
          pve-edk2-firmware-legacy | pve-edk2-firmware (<< 4~),
          pve-edk2-firmware-ovmf | pve-edk2-firmware (>= 3.20210831-1),
+         pve-edk2-firmware-aarch64,
          pve-firewall (>= 5.0.4),
          pve-ha-manager (>= 3.0-9),
          pve-qemu-kvm (>= 7.1~),
