diff --git a/debian/changelog b/debian/changelog
index dbb548be..afdaaa98 100644
--- a/debian/changelog
+++ b/debian/changelog
@@ -12,6 +12,12 @@ qemu-server (8.3.3) bookworm; urgency=medium
 
  -- Proxmox Support Team <support@proxmox.com>  Sun, 15 Dec 2024 14:26:32 +0100
 
+qemu-server (8.3.2+port) bookworm; urgency=medium
+
+  * Add port verison
+
+ -- Jiangcuo <Jiangcuo@lierfang.com>  Thu, 26 Dec 2024 11:28:33 +0800
+
 qemu-server (8.3.2) bookworm; urgency=medium
 
   * use image format from storage layer for PVE-managed volumes
