diff --git a/PVE/API2/Qemu/Machine.pm b/PVE/API2/Qemu/Machine.pm
index afb535c2..db6c3d80 100644
--- a/PVE/API2/Qemu/Machine.pm
+++ b/PVE/API2/Qemu/Machine.pm
@@ -38,7 +38,7 @@ __PACKAGE__->register_method({
 		},
 		type => {
 		    type => 'string',
-		    enum => ['q35', 'i440fx'],
+		    enum => ['q35', 'virt'],
 		    description => "The machine type.",
 		},
 		version => {
@@ -50,7 +50,7 @@ __PACKAGE__->register_method({
     },
     code => sub {
 	my $machines = eval {
-	    my $raw = file_get_contents('/usr/share/kvm/machine-versions-x86_64.json');
+	    my $raw = file_get_contents('/usr/share/kvm/machine-versions-aarch64.json');
 	    return from_json($raw, { utf8 => 1 });
 	};
 	die "could not load supported machine versions - $@\n" if $@;

diff --git a/PVE/QemuServer.pm b/PVE/QemuServer.pm
index b26da505..fc8ca388 100644
--- a/PVE/QemuServer.pm
+++ b/PVE/QemuServer.pm
@@ -4112,19 +4125,15 @@ sub config_to_command {
 	$machine_type_min =~ s/\+pve\d+$//;
 	$machine_type_min .= "+pve$required_pve_version";
     }
-    push @$machineFlags, "type=${machine_type_min}";
-
-    PVE::QemuServer::Machine::assert_valid_machine_property($conf, $machine_conf);
-
-    if (my $viommu = $machine_conf->{viommu}) {
-	if ($viommu eq 'intel') {
-	    unshift @$devices, '-device', 'intel-iommu,intremap=on,caching-mode=on';
-	    push @$machineFlags, 'kernel-irqchip=split';
-	} elsif ($viommu eq 'virtio') {
-	    push @$devices, '-device', 'virtio-iommu-pci';
+    if ($arch eq 'aarch64'){
+	if (!$kvm){
+        push @$machineFlags, "type=${machine_type_min}";
+	}else{
+	 push @$machineFlags, "type=${machine_type_min},gic-version=host";
 	}
+    }else{
+       push @$machineFlags, "type=${machine_type_min}";
     }
-
     push @$cmd, @$devices;
     push @$cmd, '-rtc', join(',', @$rtcFlags) if scalar(@$rtcFlags);
     push @$cmd, '-machine', join(',', @$machineFlags) if scalar(@$machineFlags);

