diff --git a/PVE/QemuServer/PCI.pm b/PVE/QemuServer/PCI.pm
index c4efd943..85e50eff 100644
--- a/PVE/QemuServer/PCI.pm
+++ b/PVE/QemuServer/PCI.pm
@@ -153,6 +153,7 @@ sub get_pci_addr_map {
 			ahci0 => { bus => 0, addr => 7 },
 			qga0 => { bus => 0, addr => 8 },
 			spice => { bus => 0, addr => 9 },
+			'xhci' => { bus => 0, addr => 10 }, # Some operating systems support only PCI.0 addresses, such as windows
 			# net0 will not be hotpluged, This is a legacy issue, and if the addr is modified during the update, 
 			# it may cause the guest to lose internet access.
 			# So we reserved this address for net0.
@@ -178,7 +179,6 @@ sub get_pci_addr_map {
 			hostpci13 => { bus => 18, addr => 0 },  # pcie-port 18 -> pcie.18 bus => 0, addr => 29 
 			hostpci14 => { bus => 19, addr => 0 },  # pcie-port 19 -> pcie.19 bus => 0, addr => 30 
 			'rng0' => { bus => 0, addr => 31 },
-			'xhci' => { bus => 1, addr => 1 }, # This support usbX hotplug
 			# 'net0' => { bus => 1, addr => 3 }, This support net0 hotplug
 			'net1' => { bus => 1, addr => 4 },
 			'net2' => { bus => 1, addr => 5 },
