diff --git a/PVE/QemuServer.pm b/PVE/QemuServer.pm
index b26da505..fc8ca388 100644
--- a/PVE/QemuServer.pm
+++ b/PVE/QemuServer.pm
@@ -3125,7 +3131,14 @@ sub add_tpm_device {
 
     push @$devices, "-chardev", "socket,id=tpmchar,path=$paths->{socket}";
     push @$devices, "-tpmdev", "emulator,id=tpmdev,chardev=tpmchar";
+
+	#https://bugzilla.proxmox.com/show_bug.cgi?id=4219
+	my $arch = $conf->{arch} // get_host_arch();
+	if ($arch eq 'x86_64'){
     push @$devices, "-device", "tpm-tis,tpmdev=tpmdev";
+	}else{
+	push @$devices, "-device", "tpm-tis-device,tpmdev=tpmdev";
+	}
 }
