diff --git a/PVE/API2/Qemu.pm b/PVE/API2/Qemu.pm
index 7110378e..30489608 100644
--- a/PVE/API2/Qemu.pm
+++ b/PVE/API2/Qemu.pm
@@ -17,7 +17,7 @@ use PVE::CGroup;
 use PVE::Cluster qw (cfs_read_file cfs_write_file);;
 use PVE::RRD;
 use PVE::SafeSyslog;
-use PVE::Tools qw(extract_param);
+use PVE::Tools qw(extract_param get_host_arch);
 use PVE::Exception qw(raise raise_param_exc raise_perm_exc);
 use PVE::Storage;
 use PVE::JSONSchema qw(get_standard_option);
@@ -675,6 +675,7 @@ my $hwtypeoptions = {
 
 my $generaloptions = {
     'agent' => 1,
+    'arch' => 1,
     'autostart' => 1,
     'bios' => 1,
     'description' => 1,
@@ -1225,10 +1226,10 @@ __PACKAGE__->register_method({
 		    );
 		    $conf->{$_} = $created_opts->{$_} for keys $created_opts->%*;
 
-            # set default bios to ovmf
-            if (!$conf->{bios}){
-                $conf->{bios} = 'ovmf';
-            }
+		    # set default bios to ovmf
+		    if (!$conf->{bios}){
+			$conf->{bios} = 'ovmf';
+		    }
 
 		    if (!$conf->{boot}) {
 			my $devs = PVE::QemuServer::get_default_bootdevices($conf);
@@ -1243,6 +1244,10 @@ __PACKAGE__->register_method({
 			$conf->{smbios1} = PVE::QemuServer::generate_smbios1_uuid();
 		    }
 
+		    if (!$conf->{arch}) {
+			$conf->{arch} = get_host_arch();
+		    }
+
 		    if ((!defined($conf->{vmgenid}) || $conf->{vmgenid} eq '1') && $arch ne 'aarch64') {
 			$conf->{vmgenid} = PVE::QemuServer::generate_uuid();
 		    }
diff --git a/PVE/CLI/qm.pm b/PVE/CLI/qm.pm
index 4214a7ca..18ab3326 100755
--- a/PVE/CLI/qm.pm
+++ b/PVE/CLI/qm.pm
@@ -1124,11 +1124,11 @@ our $cmddef = {
 	my $vmlist = shift;
 	exit 0 if (!scalar(@$vmlist));
 
-	printf "%10s %-20s %-10s %-10s %12s %-10s\n",
-	qw(VMID NAME STATUS MEM(MB) BOOTDISK(GB) PID);
+	printf "%10s %-20s %-12s %-10s %-10s %12s %-10s\n",
+	qw(VMID NAME ARCH STATUS MEM(MB) BOOTDISK(GB) PID);
 
 	foreach my $rec (sort { $a->{vmid} <=> $b->{vmid} } @$vmlist) {
-	    printf "%10s %-20s %-10s %-10s %12.2f %-10s\n", $rec->{vmid}, $rec->{name},
+	    printf "%10s %-20s %-12s %-10s %-10s %12.2f %-10s\n", $rec->{vmid}, $rec->{name}, $rec->{arch},
 	        $rec->{qmpstatus} || $rec->{status},
 	        ($rec->{maxmem} || 0)/(1024*1024),
 	        ($rec->{maxdisk} || 0)/(1024*1024*1024),
diff --git a/PVE/QemuServer.pm b/PVE/QemuServer.pm
index d351dfbc..302bf73d 100644
--- a/PVE/QemuServer.pm
+++ b/PVE/QemuServer.pm
@@ -3017,6 +3017,7 @@ sub vmstatus {
 	$d->{cpus} = $conf->{vcpus} if $conf->{vcpus};
 
 	$d->{name} = $conf->{name} || "VM $vmid";
+	$d->{arch}  = $conf->{arch} || get_host_arch();
 	$d->{maxmem} = get_current_memory($conf->{memory})*(1024*1024);
 
 	if ($conf->{balloon}) {
