From 974e952c2b49e3c25005a846cbd20b02429c3652 Mon Sep 17 00:00:00 2001
From: jiangcuo <jiangcuo@bingsin.com>
Date: Sat, 2 Dec 2023 05:11:40 -0500
Subject: [PATCH] update

---
 debian/control | 10 ----------
 1 file changed, 10 deletions(-)

diff --git a/debian/control b/debian/control
index d14e449..98157e9 100644
--- a/debian/control
+++ b/debian/control
@@ -14,7 +14,6 @@ Depends: apt,
          openssh-server,
          proxmox-archive-keyring,
          proxmox-kernel-helper (>= 8.0.3),
-         proxmox-default-kernel,
          pve-manager (>= 8.0.4),
          pve-qemu-kvm,
          qemu-server,
@@ -24,12 +23,3 @@ Description: Proxmox Virtual Environment
  virtualization platform for running Virtual Appliances and Virtual
  Machines. This is a meta package which will install everything
  needed.
-
-Package: pve-headers
-Architecture: all
-Depends: proxmox-default-headers,
-         ${misc:Depends},
-Section: oldlibs
-Description: Default Proxmox Kernel Headers (transitional package)
- This is a dummy transitional package to transition to proxmox-default-headers.
- It can be safely removed.
