From d8725bc9a988c81c713e17a4cdde62d23630ac3e Mon Sep 17 00:00:00 2001
From: jiangcuo <jiangcuo@bingsin.com>
Date: Mon, 16 Oct 2023 12:24:22 +0800
Subject: [PATCH] init

---
 src/mini-journalreader.c | 7 ++++---
 1 file changed, 4 insertions(+), 3 deletions(-)

diff --git a/src/mini-journalreader.c b/src/mini-journalreader.c
index 98bcaac..1396d2e 100644
--- a/src/mini-journalreader.c
+++ b/src/mini-journalreader.c
@@ -169,7 +169,8 @@ static bool print_field(sd_journal *j, const char *field) {
     if (json) {
         char tmp[7];
         for (size_t i = 0; i < l;i++) {
-            if (d[i] == '"' || d[i] == '\\' || (d[i] >= 0 && d[i] <= 0x1F)) {
+               int x = (int)d[i];
+                if (d[i] == '"' || d[i] == '\\' || (x >= 0 && x <= 0x1F)) {
                 sprintf(tmp, "\\u%04X", d[i]);
                 print_to_buf(tmp, 6);
             } else {
@@ -256,11 +257,11 @@ int main(int argc, char *argv[]) {
     const char *endcursor = NULL;
     uint64_t begin = 0;
     uint64_t end = 0;
-    char c;
+    int c;
 
     progname = argv[0];
 
-    while ((c = (char)getopt (argc, argv, "b:e:d:n:f:t:jh")) != -1) {
+while ((c = getopt (argc, argv, "b:e:d:n:f:t:jh")) != -1) {
         switch (c) {
             case 'b':
                 begin = arg_to_uint64(optarg);
