diff --git a/Makefile b/Makefile
index 758463c..aea517d 100644
--- a/Makefile
+++ b/Makefile
@@ -5,8 +5,9 @@ BUILDDIR ?= $(PACKAGE)-$(DEB_VERSION)
 DSC=$(PACKAGE)_$(DEB_VERSION).dsc
 
 GITVERSION:=$(shell git rev-parse HEAD)
+ARCH:=$(shell dpkg-architecture -qDEB_BUILD_ARCH)
 
-DEB=$(PACKAGE)_$(DEB_VERSION_UPSTREAM_REVISION)_all.deb
+DEB=$(PACKAGE)_$(DEB_VERSION_UPSTREAM_REVISION)_$(ARCH).deb
 
 all:
 
diff --git a/debian/control b/debian/control
index e04fad0..f7a202c 100644
--- a/debian/control
+++ b/debian/control
@@ -21,15 +21,15 @@ Standards-Version: 4.6.2
 Homepage: https://www.proxmox.com
 
 Package: libpve-storage-perl
-Architecture: all
+Architecture: any
 Breaks: libpve-guest-common-perl (<< 4.0-3),
         libpve-http-server-perl (<< 4.0-3),
         pve-container (<< 3.1-2),
         pve-manager (<< 5.2-12),
         qemu-server (<< 6.1-14),
 Depends: bzip2,
-         ceph-common (>= 12.2~),
-         ceph-fuse,
+         ceph-common [arm64 amd64],
+         ceph-fuse [arm64 amd64],
          cifs-utils,
          cstream,
          glusterfs-client (>= 3.4.0-2),
