diff --git a/debian/changelog b/debian/changelog
index f0481b7..d5ed5d3 100644
--- a/debian/changelog
+++ b/debian/changelog
@@ -1,3 +1,9 @@
+proxmox-archive-keyring (3.0+port1) bookworm; urgency=medium
+
+  * Add pveport key.
+
+ -- Jiangcuo <jiangcuo@bingsin.com>  Sun, 24 Dec 2023 06:28:45 +0000
+
 proxmox-archive-keyring (3.0) bookworm; urgency=medium
 
   * drop key for Debian Buster based installations
diff --git a/debian/proxmox-archive-keyring.install b/debian/proxmox-archive-keyring.install
index 05d99cb..934ebfc 100644
--- a/debian/proxmox-archive-keyring.install
+++ b/debian/proxmox-archive-keyring.install
@@ -1,3 +1,4 @@
+debian/pveport.gpg etc/apt/trusted.gpg.d/
 debian/proxmox-release-bullseye.gpg etc/apt/trusted.gpg.d/
 debian/proxmox-release-bookworm.gpg etc/apt/trusted.gpg.d/
 debian/proxmox-offline-signing-key.pub /usr/share/keyrings/
