diff --git a/PVE/APLInfo.pm b/PVE/APLInfo.pm
index 9ea206782..4f9981b05 100644
--- a/PVE/APLInfo.pm
+++ b/PVE/APLInfo.pm
@@ -205,6 +205,11 @@ sub get_apl_sources {
 	    url => "https://releases.turnkeylinux.org/pve",
 	    file => 'aplinfo.dat',
 	},
+	{
+	    host => "mirrors.apqa.cn",
+	    url => "https://mirrors.apqa.cn/proxmox",
+	    file => 'lxcs/aplinfo.dat',
+	},
     ];
 
     return $sources;
diff --git a/aplinfo/Makefile b/aplinfo/Makefile
index 56af6dfcb..55b79b1c3 100644
--- a/aplinfo/Makefile
+++ b/aplinfo/Makefile
@@ -5,7 +5,8 @@ TRUSTED_KEYS=				\
 	proxmox-release-6.x.pubkey 	\
 	proxmox-release-bullseye.pubkey	\
 	proxmox-release-bookworm.gpg	\
-	release@turnkeylinux.com.pubkey
+	release@turnkeylinux.com.pubkey \
+	mirrors.apqa.cn.pubkey
 
 all:

diff -Nur a/aplinfo/mirrors.apqa.cn.pubkey b/aplinfo/mirrors.apqa.cn.pubkey
--- a/aplinfo/mirrors.apqa.cn.pubkey
+++ b/aplinfo/mirrors.apqa.cn.pubkey
@@ -0,0 +1,52 @@
+-----BEGIN PGP PUBLIC KEY BLOCK-----
+
+mQINBGN6ANoBEAC85MMEYM2e250SzzQ5MZbec/o68oTlRNlwSkh3PcBjJYdHgAKy
+WLNpDfTRhbD1zNeHJrBtLxoBUZ8DSPQM47wfnE/4kylye3jRM0LyjUZmUSsFG36B
+F6jfxiJtQBZINaEZOR2y0eQlYUE1qdt00f7E5C/t3eUvm+GyFEgRD5bLxNkzDJDh
+7MC9VN5C2412VOorG1bOKpMpooB9PnpBnQMC/yC2FxJA3Yfnz4o/ZvwMzlJm3ytk
+d/e285mw6oe+sWLEGTF2qM+jg0W5myyDfOqBMl2+s3bvj7zmDPKvYJ8hVr5jOHQk
+OXeu5Ka7FjFaSkNkHJ82OuUOsgzzDG5r362wGNi3e7nfaWz5ukR32fI7Vi8BqcoT
+xP1V+qIDk0KsGqdakW00JZ8+O8og69LBl2YdfJKZy/dMATUFFleSF9uYRK69UjOW
+b54UTpu8PCTW0ZbqoH46sc+gZMSEostGRFAdyEv9h4rMQ+FgItPDo0b2rzSRfQLk
+Zy18bFrqdIvYZbcvZ4h14rHJjHor2dZii2Qf/utTYHhNT/HEJPYgZxMg47WAztOJ
+jVXp03owUMYyErPRq1jjaMLR++VPSGC5Z5L3J/REUHkTTRWzAx9q59h5IeSej2IB
++THb0SVlkbrB2Yd6BcPuPHD0wwX+rM+eN6R4LuWWsaSlI4X/3o8HzirkZwARAQAB
+tDhqaWFuZ2N1byAodGhpcyBpcyBmb3IgcHZlYXJtIGtleSkgPGppYW5nY3VvQGJp
+bmdzaW4uY29tPokCTgQTAQoAOBYhBDneY8fVejISR4XmPbhZUH1rH0bTBQJjegDa
+AhsDBQsJCAcCBhUKCQgLAgQWAgMBAh4BAheAAAoJELhZUH1rH0bTPTIQALvjSjxI
+0v7zeESDIPoqaO6VvO/oU7C7r6Vl6WKatD8SoTRPzjIori8UM20jLLvMWZJ2BHqv
+6BBAvpB3wnq5ypjK5DZq8incWABwfnHPfr3sSM52kiZtkB2jECJWptkmAbMrDwIk
+fwlUcPo8QzaiOjA6EhNdBEecIVdBwdt/EP1CbqgchZS89K6+6+sKjRgaTnBSxekR
+DtW33mWfqkZDPhFpIYWZeNpJi/+nSAi6EKyEdEF0yjkfkW3e7FUtia0vj19rQzJ0
+jULe9ueJrzA3UL+Nj3p7yRsDg1/iW+kWg8j7INWhnhVy7j2VXFRO69kYDBUWC45G
+nJz1TW/KEDS4Qtap+mMVkkIO6kCQYYUHmWIJjTk261CGv5EFqwZ/A/9KBzs0GfPr
+swpUAx+rapHosDRFVVIj7rvzrR3axQ3tEGvg8I2VmEj1GHYukhbhdfmFhYiS5EjP
+rTTos8szjAJoyYRYzqWHN5YF4ur2Kddcaoqggtq/YIpaLG8R/VSxktJpK1fRNCvA
+JOtGNd2PQweGnSWsya/emQjr4VIlFRN4e1OIkwlf8nI83Qc66AorIasEOjLa5fo1
+tDsTjNmGnOLYM2bXnfJUK6Yzpolie2dtK0ACgD6YEkcvhV4FzWUXkZn1I9MvzENu
+V3WQm2aMaKN6hb5zjv6Rs2QXEX0w55YLjTYUuQINBGN6ANoBEADgBCW+XtBhi/qZ
+Svhp5W86MiHw82sB/N9u/CV7AHM1jKc6gbN5DlgDXm5liHQwNEvnuoaZIwWIiFmC
+rWrm2me2ACHqOlZJsDoVwzTBpDH1belznBzJ2COFU0eEGShBNLiVasGUMq/9waLX
+4g5DTxACa7xahVsT8S9LRb/malFtBa+xpab30ccyu6OeDnEc2Kq3oFF87xQ8Yl+W
+CJUAW3txHcjMCCTmwcmpQ2UoV5Bv7BQsJjntGZUA88lKv4gbl36jtlkvDUfXcZwS
+e6MZUDDFrr2d3lBtuMSJSBPabq336r1UC94rL0HjDlLhnu2AMrk5OkMEwcNYV65W
+aX4J+/jdSz1NQBQb5J28DdgRQwlTqM9QIn5ahIFtj+vSmfRjIrqp69ntjEYwzDMT
+dhVD5ZyJkAwJotnb0zsO7InRm1pG8mrMBsSBiy0cn4P1xSCDWH5aMs8uZs8FPauG
+rZ+nhyW97aA41r6azDUUFMCmXR4Xylm+aNKUle/gh8sZkR3Gbph5+vVfRCBiKMIZ
+TIyDKt2i74/VRUoibtpaVqcgvpECfdffPKyi9vMX3fbZg5KLtSUS8ndhI/oWspDi
+vNhpNjUTiB/vmUco5UR+1gZxkhzuXlvLDLrYYmj9TaBV77cYBgYs+di1oy3gvoCw
+TGkYpweKAbwDkbKAa/AruQjv0WnkAQARAQABiQI2BBgBCgAgFiEEOd5jx9V6MhJH
+heY9uFlQfWsfRtMFAmN6ANoCGwwACgkQuFlQfWsfRtPsRQ/9GG6dCbKZKpIupChJ
+oT3H6Z/v7lbXNwIdAsA0yK8W29PFjCdy+EGgXnCviwUrqoVoH1QttZRIj9bT8Q8a
+qnxgQA/ZRypcZdh1TvXTiOzW1wDp5XmDuZnNACAO16mqQww7nnA38Nfcggyy9LMd
+F3jnJ6dFPkQTk3bBPVSLO4p7d7DOOVy9fkkBbT5y1tlPHUJp/IGGwAjdSoLMjwps
+Su+CePlaOvIoTXMdibkoL/fwAr+38ha2dJ3nKyKkQlZyYzkdWjefEyguEo4tmrHD
+4aTKwJitEzLUbprqhgP1+diq9zfc6uBy3bXMn6kU5u4ZByJF8fhapJzQkttq2m4H
+iybPGkLLqvV3Ovbw8OUmnR+czU/dAO4yCUMrsr/PPsg4xf7geh9BDyZWkJRmuzI8
+sCOdkSdr81kLg6hDmxtIKesoxMjvTL8ZuD4jU78yROBJDpkO9X0TeC0lFmGSDlt+
+/84vJH+FCKd/QwDxHe1dChc0lIDFl/64K9FjCTOwmZ6UpZ/0xNAMY26sqAeVQ9h3
+z9m7oPZCetlF3YpuXE6yIKYJaKq2MUkWjdevrhrQq72TlA9YrLLNJbiseK8T4A8S
+Zomcx8slMe+O8IedvEw5lR9rgSJ/nJZNy0+u3fBUqhyJoYLRzYWGeUDkHsu6RpTo
+wr7Zz10/ZojX6eOcRH3kIslX9Lc=
+=BaeF
+-----END PGP PUBLIC KEY BLOCK-----
