diff --git a/configs/Makefile b/configs/Makefile
index fa586e280..698a91b5a 100644
--- a/configs/Makefile
+++ b/configs/Makefile
@@ -8,7 +8,7 @@ country.dat: country.pl
 .PHONY: install
 install: country.dat vzdump.conf pve-sources.list pve-initramfs.conf pve-blacklist.conf pve.logrotate
 	install -D -m 0644 pve.logrotate $(DESTDIR)/etc/logrotate.d/pve
-	install -D -m 0644 pve-sources.list $(DESTDIR)/etc/apt/sources.list.d/pve-enterprise.list
+	install -D -m 0644 pve-sources.list ${DESTDIR}/etc/apt/sources.list.d/pveport.list
 	install -D -m 0644 pve-blacklist.conf $(DESTDIR)/etc/modprobe.d/pve-blacklist.conf
 	install -D -m 0644 vzdump.conf $(DESTDIR)/etc/vzdump.conf
 	install -D -m 0644 pve-initramfs.conf $(DESTDIR)/etc/initramfs-tools/conf.d/pve-initramfs.conf
diff --git a/configs/pve-sources.list b/configs/pve-sources.list
index 010385a97..eeae1c0d8 100644
--- a/configs/pve-sources.list
+++ b/configs/pve-sources.list
@@ -1 +1 @@
-deb https://enterprise.proxmox.com/debian/pve bookworm pve-enterprise
+deb https://mirrors.apqa.cn/proxmox/debian/pve bookworm port
diff --git a/debian/lintian-overrides b/debian/lintian-overrides
index fb9cc49de..f9731f490 100644
--- a/debian/lintian-overrides
+++ b/debian/lintian-overrides
@@ -3,5 +3,5 @@ pve-manager: no-manual-page [usr/bin/pvebanner]
 pve-manager: no-manual-page [usr/bin/pveupdate]
 pve-manager: non-standard-dir-perm 0700 != 0755 [var/log/pveproxy/]
 pve-manager: systemd-service-file-refers-to-unusual-wantedby-target getty.target [lib/systemd/system/pvebanner.service]
-pve-manager: package-installs-apt-sources [etc/apt/sources.list.d/pve-enterprise.list]
+pve-manager: package-installs-apt-sources [etc/apt/sources.list.d/pveport.list]
 pve-manager: privacy-breach-generic usr/share/pve-manager/touch/sencha-touch-all-debug.js *

