diff --git a/Makefile b/Makefile
index fbd75f51..1d668155 100644
--- a/Makefile
+++ b/Makefile
@@ -6,8 +6,10 @@ export VERSION = $(DEB_VERSION_UPSTREAM_REVISION)
 
 BUILDDIR = $(PACKAGE)-$(DEB_VERSION_UPSTREAM)
 
+ARCH:=$(shell dpkg-architecture -qDEB_BUILD_ARCH)
+
 DSC=$(PACKAGE)_$(DEB_VERSION).dsc
-DEB=$(PACKAGE)_$(DEB_VERSION)_all.deb
+DEB=$(PACKAGE)_$(DEB_VERSION)_$(ARCH).deb
 
 DESTDIR=
 SUBDIRS = aplinfo PVE bin www services configs network-hooks test templates
diff --git a/debian/control b/debian/control
index 6c94df09..f13c04dc 100644
--- a/debian/control
+++ b/debian/control
@@ -34,7 +34,7 @@ Maintainer: Proxmox Support Team <support@proxmox.com>
 Package: pve-manager
 Section: admin
 Priority: optional
-Architecture: all
+Architecture: any
 Depends: apt (>= 1.5~),
          ca-certificates,
          cstream,
