diff --git a/Makefile b/Makefile
index 92fc0596..54601ee2 100644
--- a/Makefile
+++ b/Makefile
@@ -58,8 +58,6 @@ install: vzdump-hook-script.pl
 	install -d $(DESTDIR)/var/lib/vz/template/iso
 	install -m 0644 vzdump-hook-script.pl $(DOCDIR)/examples/vzdump-hook-script.pl
 	install -m 0644 spice-example-sh $(DOCDIR)/examples/spice-example-sh
-	install -d $(DESTDIR)/lib/udev/rules.d
-	install -m 0644 udev/99-query-machine-info.rules $(DESTDIR)/lib/udev/rules.d/99-query-machine-info.rules
 	set -e && for i in $(SUBDIRS); do $(MAKE) -C $$i $@; done
 
 .PHONY: distclean
diff --git a/udev/99-query-machine-info.rules b/udev/99-query-machine-info.rules
deleted file mode 100644
index dc28d11b..00000000
--- a/udev/99-query-machine-info.rules
+++ /dev/null
@@ -1,2 +0,0 @@
-ACTION=="add", SUBSYSTEM=="*", RUN="/usr/bin/query-machine-info"
-ACTION=="remove", SUBSYSTEM=="*", RUN="/usr/bin/query-machine-info"
