--- a/purgatory/Makefile
+++ b/purgatory/Makefile
@@ -67,7 +67,7 @@
 $(PURGATORY): $(PURGATORY_OBJS)
 	$(MKDIR) -p $(@D)
 	$(CC) $(CFLAGS) $(LDFLAGS) -o $@.sym $^
-#	$(LD) $(LDFLAGS) $(EXTRA_LDFLAGS) --no-undefined -e purgatory_start -r -o $@ $(PURGATORY_OBJS) $(UTIL_LIB)
+#	$(LD) $(LDFLAGS) $(EXTRA_LDFLAGS) -Wl,--no-undefined -e purgatory_start -r -o $@ $(PURGATORY_OBJS) $(UTIL_LIB)
 	$(STRIP) --strip-debug -o $@ $@.sym
 
 echo::
