diff --git a/proxinstall b/proxinstall
index 12f3eaa..dcf6e7e 100755
--- a/proxinstall
+++ b/proxinstall
@@ -1592,6 +1592,6 @@ sub create_intro_view {
-    if ($iso_env->{product} eq 'pve' && !$run_env->{hvm_supported}) {
-	Proxmox::UI::error(
-	    "No support for hardware-accelerated KVM virtualization detected.\n\n"
-	    ."Check BIOS settings for Intel VT / AMD-V / SVM."
-	);
-    }
+#    if ($iso_env->{product} eq 'pve' && !$run_env->{hvm_supported}) {
+#	Proxmox::UI::error(
+#	    "No support for hardware-accelerated KVM virtualization detected.\n\n"
+#	    ."Check BIOS settings for Intel VT / AMD-V / SVM."
+#	);
+#   }
diff --git a/proxmox-tui-installer/src/main.rs b/proxmox-tui-installer/src/main.rs
index 3fb87a7..242978b 100644
--- a/proxmox-tui-installer/src/main.rs
+++ b/proxmox-tui-installer/src/main.rs
@@ -207,15 +207,15 @@ fn installer_setup_late(siv: &mut Cursive) {
         );
     }
 
-    if state.setup_info.config.product == ProxmoxProduct::PVE && !state.runtime_info.hvm_supported {
-        display_setup_warning(
-            siv,
-            concat!(
-                "No support for hardware-accelerated KVM virtualization detected.\n\n",
-                "Check BIOS settings for Intel VT / AMD-V / SVM."
-            ),
-        );
-    }
+    // if state.setup_info.config.product == ProxmoxProduct::PVE && !state.runtime_info.hvm_supported {
+    //     display_setup_warning(
+    //         siv,
+    //         concat!(
+    //             "No support for hardware-accelerated KVM virtualization detected.\n\n",
+    //             "Check BIOS settings for Intel VT / AMD-V / SVM."
+    //         ),
+    //     );
+    // }
 }
 
 fn initial_setup_error(siv: &mut CursiveRunnable, message: &str) -> ! {
