diff --git a/debian/control b/debian/control
index 04b0c6e..c6232af 100644
--- a/debian/control
+++ b/debian/control
@@ -34,14 +34,22 @@ Homepage: https://www.proxmox.com
 
 Package: proxmox-installer
 Architecture: any
-Depends: chrony,
+Depends: btrfs-progs,
+         chrony,
+         eject,
          geoip-bin,
          iproute2,
          libgtk3-perl,
          libgtk3-webkit2-perl,
          libjson-perl,
+         lvm2,
+         openbox,
          proxmox-kernel-helper,
          squashfs-tools,
+         thin-provisioning-tools,
+         xorg,
+         zfsutils-linux,
+         zfs-initramfs,
          ${misc:Depends},
          ${perl:Depends},
          ${shlibs:Depends},
