diff --git a/Proxmox/Sys/Block.pm b/Proxmox/Sys/Block.pm
index e337474..cb728cb 100644
--- a/Proxmox/Sys/Block.pm
+++ b/Proxmox/Sys/Block.pm
@@ -171,6 +171,8 @@ sub get_partition_dev {
 	return "${dev}$partnum";
     } elsif ($dev =~ m|^/dev/nvme\d+n\d+$|) {
 	return "${dev}p$partnum";
+    } elsif ($dev =~ m|^/dev/mmcblk\d+$|) {
+        return "${dev}p$partnum"; 
     } else {
 	die "unable to get device for partition $partnum on device $dev\n";
     }
