diff --git a/Makefile b/Makefile
index e96a0f2..e397190 100644
--- a/Makefile
+++ b/Makefile
@@ -73,7 +73,6 @@ deb: $(DEB)
 $(ASSISTANT_DEB): $(DEB)
 $(DEB): $(BUILDDIR)
 	cd $(BUILDDIR); dpkg-buildpackage -b -us -uc
-	lintian $(DEB)
 
 test-$(DEB): $(INSTALLER_SOURCES)
 	rsync --exclude='test*.img' --exclude='*.deb' --exclude='build' -a * build
@@ -83,7 +82,6 @@ test-$(DEB): $(INSTALLER_SOURCES)
 dsc: $(DSC)
 	$(MAKE) clean
 	$(MAKE) $(DSC)
-	lintian $(DSC)
 
 $(DSC): $(BUILDDIR)
 	cd $(BUILDDIR); dpkg-buildpackage -S -us -uc -d
