diff --git a/pixman/pixman-arma64-neon-asm.S b/pixman/pixman-arma64-neon-asm.S
index 7329d4b..1c1fdd5 100644
--- a/pixman/pixman-arma64-neon-asm.S
+++ b/pixman/pixman-arma64-neon-asm.S
@@ -1325,7 +1325,7 @@ generate_composite_function \
 .macro pixman_composite_src_8888_8888_process_pixblock_tail_head
     st1  {v0.2s, v1.2s, v2.2s, v3.2s}, [DST_W], #32
     fetch_src_pixblock
-    cache_preload 8, 8
+    /* cache_preload 8, 8 */
 .endm

 generate_composite_function \
