diff --git a/Makefile b/Makefile
index bd38c95..cbf47f6 100644
--- a/Makefile
+++ b/Makefile
@@ -36,6 +36,11 @@ fw.list: fwlist-6.2.2-1-pve
 fw.list: fwlist-6.2.6-1-pve
 fw.list: fwlist-6.2.16-11-pve
 fw.list: fwlist-6.5.3-1-pve
+fw.list: fwlist-6.1-generic
+fw.list: fwlist-6.6-openeuler
+fw.list: fwlist-6.6-openeuler-loong64
+fw.list: fwlist-6.6-phytium
+fw.list: fwlist-6.11.0-loong64
 fw.list: fwlist-6.8.1-1-pve
 fw.list: fwlist-6.8.8-1-pve
 fw.list: fwlist-6.8.12-1-pve
