diff --git a/Makefile b/Makefile
index 82630da..8c47b77 100644
--- a/Makefile
+++ b/Makefile
@@ -1,29 +1,31 @@
 include /usr/share/dpkg/default.mk
 # source form https://github.com/zfsonlinux/
 
+ARCH := $(shell dpkg-architecture -qDEB_HOST_ARCH)
+
 PACKAGE = zfs-linux
 
 SRCDIR = upstream
 BUILDDIR ?= $(PACKAGE)-$(DEB_VERSION_UPSTREAM)
 ORIG_SRC_TAR = $(PACKAGE)_$(DEB_VERSION_UPSTREAM).orig.tar.gz
 
-ZFS_DEB1= libnvpair3linux_$(DEB_VERSION)_amd64.deb
+ZFS_DEB1= libnvpair3linux_$(DEB_VERSION)_$(ARCH).deb
 
 ZFS_DEB_BINARY =				\
-libpam-zfs_$(DEB_VERSION)_amd64.deb		\
-libuutil3linux_$(DEB_VERSION)_amd64.deb		\
-libzfs4linux_$(DEB_VERSION)_amd64.deb		\
-libzfsbootenv1linux_$(DEB_VERSION)_amd64.deb	\
-libzpool5linux_$(DEB_VERSION)_amd64.deb		\
-zfs-test_$(DEB_VERSION)_amd64.deb			\
-zfsutils-linux_$(DEB_VERSION)_amd64.deb		\
-zfs-zed_$(DEB_VERSION)_amd64.deb
+libpam-zfs_$(DEB_VERSION)_$(ARCH).deb		\
+libuutil3linux_$(DEB_VERSION)_$(ARCH).deb		\
+libzfs4linux_$(DEB_VERSION)_$(ARCH).deb		\
+libzfsbootenv1linux_$(DEB_VERSION)_$(ARCH).deb	\
+libzpool5linux_$(DEB_VERSION)_$(ARCH).deb		\
+zfs-test_$(DEB_VERSION)_$(ARCH).deb			\
+zfsutils-linux_$(DEB_VERSION)_$(ARCH).deb		\
+zfs-zed_$(DEB_VERSION)_$(ARCH).deb
 
-ZFS_DBG_DEBS = $(patsubst %_$(DEB_VERSION)_amd64.deb, %-dbgsym_$(DEB_VERSION)_amd64.deb, $(ZFS_DEB1) $(ZFS_DEB_BINARY))
+ZFS_DBG_DEBS = $(patsubst %_$(DEB_VERSION)_$(ARCH).deb, %-dbgsym_$(DEB_VERSION)_$(ARCH).deb, $(ZFS_DEB1) $(ZFS_DEB_BINARY))
 
 ZFS_DEB2= $(ZFS_DEB_BINARY)			\
-libzfslinux-dev_$(DEB_VERSION)_amd64.deb		\
-python3-pyzfs_$(DEB_VERSION)_amd64.deb		\
+libzfslinux-dev_$(DEB_VERSION)_$(ARCH).deb		\
+python3-pyzfs_$(DEB_VERSION)_$(ARCH).deb		\
 pyzfs-doc_$(DEB_VERSION)_all.deb			\
 spl_$(DEB_VERSION)_all.deb			\
 zfs-initramfs_$(DEB_VERSION)_all.deb
@@ -62,7 +64,12 @@ zfs: $(DEBS)
 $(ZFS_DEB2) $(ZFS_DBG_DEBS): $(ZFS_DEB1)
 $(ZFS_DEB1): $(BUILDDIR)
 	cd $(BUILDDIR); dpkg-buildpackage -b -uc -us
+ifeq ($(ARCH), loong64)
+	echo "Architecture is loongarch64. Skipping lintian."
+else
 	lintian $(DEBS)
+endif
+
 
 $(ORIG_SRC_TAR): $(BUILDDIR)
 	tar czf $(ORIG_SRC_TAR) --exclude="$(BUILDDIR)/debian" $(BUILDDIR)
