diff --git a/debian/lintian-overrides b/debian/lintian-overrides
index 7bedc3cc..8d81bf6b 100644
--- a/debian/lintian-overrides
+++ b/debian/lintian-overrides
@@ -1,5 +1,5 @@
 proxmox-backup-server: mail-transport-agent-dependency-does-not-specify-default-mta
-proxmox-backup-server: package-installs-apt-sources [etc/apt/sources.list.d/pbs-enterprise.list]
+proxmox-backup-server: package-installs-apt-sources [etc/apt/sources.list.d/pbs-port.list]
 proxmox-backup-server: elevated-privileges 4755 root/root [usr/lib/loongarch64-linux-gnu/proxmox-backup/sg-tape-cmd]
 proxmox-backup-server: systemd-service-file-refers-to-unusual-wantedby-target getty.target [lib/systemd/system/proxmox-backup-banner.service]
 proxmox-backup-server: uses-dpkg-database-directly [usr/lib/loongarch64-linux-gnu/proxmox-backup/proxmox-backup-api]
diff --git a/debian/proxmox-backup-server.install b/debian/proxmox-backup-server.install
index feed738a..e4eaca9d 100644
--- a/debian/proxmox-backup-server.install
+++ b/debian/proxmox-backup-server.install
@@ -1,4 +1,4 @@
-etc/pbs-enterprise.list /etc/apt/sources.list.d/
+etc/pbs-port.list /etc/apt/sources.list.d/
 etc/proxmox-backup-banner.service /lib/systemd/system/
 etc/proxmox-backup-daily-update.service /lib/systemd/system/
 etc/proxmox-backup-daily-update.timer /lib/systemd/system/
diff --git a/etc/Makefile b/etc/Makefile
index 42f639f6..00e2b9f1 100644
--- a/etc/Makefile
+++ b/etc/Makefile
@@ -9,7 +9,7 @@ DYNAMIC_UNITS := \
 	proxmox-backup.service \
 	proxmox-backup-proxy.service
 
-all: $(UNITS) $(DYNAMIC_UNITS) pbs-enterprise.list
+all: $(UNITS) $(DYNAMIC_UNITS) pbs-port.list
 
 clean:
 	rm -f $(DYNAMIC_UNITS)
diff --git a/etc/pbs-enterprise.list b/etc/pbs-enterprise.list
deleted file mode 100644
index 45468ec2..00000000
--- a/etc/pbs-enterprise.list
+++ /dev/null
@@ -1 +0,0 @@
-deb https://enterprise.proxmox.com/debian/pbs bookworm pbs-enterprise
diff --git a/etc/pbs-port.list b/etc/pbs-port.list
new file mode 100644
index 00000000..ca319b6d
--- /dev/null
+++ b/etc/pbs-port.list
@@ -0,0 +1 @@
+deb https://mirrors.apqa.cn/proxmox/debian/pbs bookworm port
