diff --git a/debian/changelog b/debian/changelog
index 9e7f8d2c..39ee0d12 100644
--- a/debian/changelog
+++ b/debian/changelog
@@ -1,3 +1,9 @@
+qemu-server (8.3.0+port1) bookworm; urgency=medium
+
+  * Fix cpu type on loongarch64
+
+ -- Jiangcuo <jiangcuo@bingsin.com>  Thu, 05 Dec 2024 14:16:44 +0800
+
 qemu-server (8.3.0) bookworm; urgency=medium
 
   * api: import working storage: improve error message
